frontend/public/components/monitoring/query-browser.tsx

```diff
@@ -673,11 +676,12 @@ const QueryBrowser_: React.FC<QueryBrowserProps> = ({
     setXDomain([from, to]);
     setSpan(to - from);
   };
+  const isGraphDataEmpty = _.isEmpty(_.flattenDeep(graphData));
```

Comment:
I think we should avoid using `_.flattenDeep` because there can be a large amount of data in `graphData` and we already have performance concerns.

Reply:
Maybe
```
const isGraphDataEmpty = _.isEmpty(graphData) || _.every(graphData, (d) => d.length === 0);
```
or
```
const isGraphDataEmpty = !_.some(graphData, (d) => d.length > 0);
```

Reply:
we could use `const isGraphDataEmpty = !graphData || graphData.every((d) => d.length === 0)`
since `every()` always returns true for any condition when array is empty, we could drop the check `_.isEmpty`
updated the checks. please take a look.

Reply:
👍 Thanks